vpr/src/place/place.cpp

```diff
@@ -2479,25 +2003,26 @@ static void alloc_and_load_placement_structs(float place_cost_exp,
     if (placer_opts.place_algorithm.is_timing_driven()) {
         /* Allocate structures associated with timing driven placement */
         /* [0..cluster_ctx.clb_nlist.nets().size()-1][1..num_pins-1]  */
-        connection_delay = make_net_pins_matrix<float>(cluster_ctx.clb_nlist, 0.f);
-        proposed_connection_delay = make_net_pins_matrix<float>(cluster_ctx.clb_nlist, 0.f);
 
-        connection_setup_slack = make_net_pins_matrix<float>(cluster_ctx.clb_nlist, std::numeric_limits<float>::infinity());
+        p_timing_ctx.connection_delay = make_net_pins_matrix<float>(cluster_ctx.clb_nlist, 0.f);
```

Comment:
I'd measure speed regularly on this refactoring to watch for slowdowns.  p_timing_ctx is a global, so if the compiler can't figure out if someone else might be changing the pointer value in connection_delay (called pointer aliasing --maybe some routine has made a copy of p_timing_ctx.connection_delay and is updating the values or the connection_delay pointer itself when you call a subroutine).  If the compiler thinks there is some chance that some routine being called could change some pointer you're using, it will reload the pointer a lot, slowing things down some.

This means sometimes making a local copy of a pointer and using it to fill in or access a data structure can be faster.  It's very hard to reason about because you're guessing what the compiler can and can't do, but it's why it's good to measure speed a lot.  Innocuous seeming changes from a local variable to a larger scope variable can slow things down due to conservative reloading of values by the compiler.

Reply:
This might become an issue when Dusty designs the parallel execution placer. According to the benchmark results so far everything is good.

Reply:
That's good news, thanks.